### ISSUE TITLE
VSTS Agent template stalls until timeout lacking latest NuGet Provider

### ISSUE THREAD

**Issue by colinbeales** [Azure/azure-quickstart-templates#7979]
201-vm-vsts-agent(https://github.com/Azure/azure-quickstart-templates/tree/master/201-vm-vsts-agent)

### Issue Details

When running the template all the resources will create and the extension will trigger the InstallVstsAgent.ps1 script to execute. This script will get as far as creating and configuring the agent and saving the modules in C:\Modules. In the next section it will stall and never return until the script hits the ARM time-out. 

I have spun up the VM and run interactively the same InstallVstsAgent.ps1 script and can see the loop around the 3 default modules will each fail with stall with an interactive prompt, which I guess is where the non-interactive execution fails having put in additional tracing through execution. Denying each interactive prompt gives the error...

```
"Find-Module : NuGet provider is required to interact with NuGet-based repositories. Please ensure that '2.8.5.201' or newer version of NuGet provider is installed"
```

### Repro steps (*if necessary, delete otherwise*)
1. Run the template under the CLI e.g. with these inputs
```
az deployment group create -g $RG_DEVOPSAGENT --name agentdeployment \
  --template-uri "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-vsts-agent/azuredeploy.json" \
  --parameters  publicIPDnsName=$(echo $PREFIX)agent \
                _artifactsLocation="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-vsts-agent/" \
                vmAdminUser=$ADO_VMADMINUSER \
                vmAdminPassword=$ADO_VMADMINPASSWORD \
                vmSize=Standard_D1_v2 \
                vstsAccount=$ADO_ACCOUNTNAME \
                vstsPersonalAccessToken=$ADO_PERSONALACCESSTOKEN \
                vstsAgentCount=1 \
                vstsPoolName=Default
```

2. CLI will create the VM, Network etc... the custom extension will trigger and start the agent install and config, but will stall at the 'Install-Module' of '"PowerShellGet", "PackageManagement","Pester"' due to lacking NuGet provider at the right level.

**Comment by colinbeales** [comment]
Submitted #7980 to fix this

**Comment by colinbeales** [comment]
Fixed in #7980
